### PR TITLE
[Perf] Minimax-H3 support fused RMSNorm and RoPE opt

### PR DESCRIPTION
**Pull request by fan2956**
PLEASE FILL IN THE PR DESCRIPTION HERE.

## Purpose
Enable MiniMax-H3 to use the fused RMSNorm and RoPE implementations provided by the Omni diffusion layers.
- Add configurable weight dtype support to Omni diffusion `RMSNorm`, while preserving fp32 accumulation in the native fallback.
- Extend Omni diffusion `RotaryEmbedding` to support full-head-dimension `cos`/`sin` inputs required by MiniMax-H3.

## Test Plan
```
MODEL="/data/weights/h3/MiniMax-H3/FL2VA"
vllm serve "${MODEL}" \
  --omni \
  --host 0.0.0.0 \
  --port "${PORT}" \
  --trust-remote-code \
  --num-gpus 8 \
  --usp 8 \
  --ring 1 \
  --text-encoder-tp-size 8 \
  --enable-layerwise-offload \
  --vae-parallel-mode tile \
  --vae-use-tiling \
  --vae-patch-parallel-size 8 \
  --diffusion-attention-backend FLASH_ATTN 
```
prompt："A little girl grows up."
image: 
first_frame

**vLLM Version:**
0.26.0

**vLLM-Omni Commit:**
5215e03a91adecbb5ffece29aa74360a7569d0c5

## Test Result
**Performance:**

E2E Latency: 491014ms -> 475070ms.      **-3.35%**

ori: 
Pasted Graphic 1

opt:
Pasted Graphic

**Precision:**
ori (left) vs opt (right)

https://github.com/user-attachments/assets/3cbd06eb-8e77-4106-a42a-eba346f71873

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: tests/diffusion/models/minimax_h3/test_minimax_h3_parallel.py [hunk 1]
```diff
@@ -61,6 +61,32 @@ def test_packed_attention_is_a_regional_compile_boundary():
...
+        MiniMaxH3Attention,
+    )
+
+    attention = object.__new__(MiniMaxH3Attention)
+    nn.Module.__init__(attention)
+    attention.rope = RotaryEmbedding(is_neox_style=True, half_head_dim=False)
+    attention.rope._forward_method = attention.rope.forward_native
+
+    x = torch.randn(11, 3, 128, dtype=torch.bfloat16)
+    freqs = torch.randn(11, 96)
```
Comment (david6666666) [inline tests/diffusion/models/minimax_h3/test_minimax_h3_parallel.py]: This test fails as written: `torch.randn(11, 96)` is not tiled, but the fused path canonicalizes cos/sin to the half-dim layout (truncate to `[..., :48]` + block-tile via `_prepare_half_head_dim_cos_sin`/`apply_rotary_emb_torch`), while the reference below multiplies each of the 96 dims by its own cos/sin. For random freqs the two differ on the second half (up to ~5 abs diff), so `assert_close(atol=0, rtol=0)` fails; they only agree for tiled freqs like `MiniMaxH3Rope`'s `cat((half, half))`. Build freqs as `torch.cat((freqs, freqs), dim=-1)` or compute the reference from the half layout.

**Comment by david6666666** [comment]
PR has merge conflicts with main (vllm_omni/diffusion/models/minimax_h3/minimax_h3_transformer.py), please rebase.

File: tests/diffusion/models/minimax_h3/test_minimax_h3_parallel.py (hunk 1 above)
Comment (fan2956) [reply tests/diffusion/models/minimax_h3/test_minimax_h3_parallel.py]: Thanks for catching this. You are right: the test must use MiniMax-H3's tiled full-dimension frequency layout.

I restored the coverage and now construct `freqs` as `cat((freqs_half, freqs_half), dim=-1)`, matching `MiniMaxH3Rope`. The reference and fused/native path now use the same RoPE semantics.

**Review by david6666666 — APPROVED** [review]
LGTM

**Comment by david6666666** [comment]
is this pr ready now?

**Comment by fan2956** [comment]
> is this pr ready now?
Not yet，main H3 infer failed on npu because #5703，this pr will verify again after this bug fix
